contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -406,4 +500,81 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
   function isCapabilityDeprecated(bytes32 hashedCapabilityId) external view returns (bool) {
     return s_deprecatedHashedCapabilityIds.contains(hashedCapabilityId);
   }
+
+  /// @notice Adds a DON made up by a group of nodes that support a list
+  /// of capability configurations
+  /// @param nodes The nodes making up the DON
+  /// @param capabilityConfigurations The list of configurations for the
+  /// capabilities supported by the DON
+  /// @param isPublic True if the DON is public
+  function addDON(
+    bytes32[] calldata nodes,
+    CapabilityConfiguration[] calldata capabilityConfigurations,
+    bool isPublic
+  ) external onlyOwner {
+    uint32 id = s_donId;
+
+    s_dons[id].id = id;
+    s_dons[id].isPublic = isPublic;
+
+    for (uint256 i; i < nodes.length; ++i) {
+      bytes32 nodeP2PId = nodes[i];
+      if (s_dons[id].nodes.contains(nodeP2PId)) revert DuplicateDONNode(id, nodeP2PId);
+
+      s_dons[id].nodes.add(nodeP2PId);
+    }
+
+    for (uint256 i; i < capabilityConfigurations.length; ++i) {
+      CapabilityConfiguration calldata configuration = capabilityConfigurations[i];
+      bytes32 capabilityId = configuration.capabilityId;
+
+      if (!s_hashedCapabilityIds.contains(capabilityId)) revert CapabilityDoesNotExist(capabilityId);
+      if (s_deprecatedHashedCapabilityIds.contains(capabilityId)) revert CapabilityIsDeprecated(capabilityId);
+
+      if (s_dons[id].capabilityIds.contains(capabilityId)) revert DuplicateDONCapability(id, capabilityId);
+
+      for (uint256 j; j < nodes.length; ++j) {
+        bytes32 nodeP2PId = nodes[j];
+        if (!s_nodes[nodeP2PId].supportedCapabilityIds[s_nodes[nodeP2PId].configCount].contains(capabilityId))
+          revert NodeDoesNotSupportCapability(nodeP2PId, capabilityId);
+      }
+
+      s_dons[id].capabilityIds.add(capabilityId);
+      s_dons[id].capabilityConfigs[capabilityId] = configuration.config;
+    }
+
+    ++s_donId;
+
+    emit DONAdded(id, isPublic);
+  }
+
+  /// @notice Get's a DON's data
+  /// @param donId The DON's ID
+  /// @return uint32 The DON ID
+  /// @return bool True if the DON is public
+  /// @return bytes32[] The list of node P2P IDs that are in the DON
+  /// @return CapabilityConfiguration[] The list of capability configurations supported by the DON
```

Comment:
```suggestion
```suggestion
  /// @notice Gets DON's data
  /// @param donId The DON ID
  /// @return bool True if the DON is public
  /// @return bytes32[] The list of node P2P IDs that are in the DON
  /// @return CapabilityConfiguration[] The list of capability configurations supported by the DON
```